frontend/public/components/dashboard/dashboards-page/cluster-dashboard/status-card.tsx

```diff
@@ -141,6 +143,15 @@ export const StatusCard = connect<StatusCardProps>(mapStateToProps)(({ k8sModels
         Component: <URLHealthItem subsystem={subsystem.properties} models={k8sModels} />,
       });
     } else if (isDashboardsOverviewHealthPrometheusSubsystem(subsystem)) {
+      const { disallowedProvider } = subsystem.properties;
+      if (disallowedProvider) {
+        if (
+          !infrastructureLoaded ||
+          getInfrastructurePlatform(infrastructure) === disallowedProvider
+        ) {
```

Comment:
This can be a single `if`

```suggestion
      if (disallowedProvider && (
          !infrastructureLoaded ||
          getInfrastructurePlatform(infrastructure) === disallowedProvider
        ) {
```